TemplateBicep/modules/observability.bicep

```diff
@@ -0,0 +1,55 @@
+param prefix string
+param iotHubName string
+param location string
+
+resource iotHub 'Microsoft.Devices/IotHubs@2021-07-02' existing = {
+  name: iotHubName
+}
+
+resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2022-10-01' = {
+  name: '${prefix}analytics'
+  location: location
+  properties: {
+    retentionInDays: 30
+    sku: {
+      name: 'PerGB2018'
+    }
+    workspaceCapping: {
+      dailyQuotaGb: 10
+    }
+    features: {
+      searchVersion: 1
+    }
+  }
+}
+
+resource appInsight 'Microsoft.Insights/components@2020-02-02' = {
+  name: '${prefix}insight'
+  kind: 'web'
+  location: location
+  properties: {
+    WorkspaceResourceId: logAnalytics.id
+    Application_Type: 'web'
+  }
+}
+
+module workbook './workbook.bicep' = {
```

Comment:
Why having a subtemplate for one resource here?

Reply:
For now, I will leave each sub-resource in own file. Let's revisit once we start working more with them.